src/methods/aggregate.jl

```diff
@@ -136,40 +142,36 @@ function aggregate!(loci::Tuple{Locus,Vararg}, dst::AbstractRaster, src, scale;
         val = src[(upsample.(Tuple(I), intscale) .+ offsets)...]
         val === missingval(src) ? missingval(dst) : val
     end
+    return dst
 end
 # Function/functor methods
-function aggregate!(f, dst::AbstractRaster, src, scale; skipmissingval=false)
+function aggregate!(f, dst::AbstractRaster, src, scale;
+    skipmissingval=false, skipmissing=skipmissingval, progress=true
+)
     intscale = _scale2int(Ag(), dims(src), scale)
-    broadcast!(dst, CartesianIndices(dst)) do I
+    # len = prod(intscale)
+    l = upsample.(map(firstindex, axes(dst)), intscale)
+    u = upsample.(map(lastindex, axes(dst)), intscale)
+    checkbounds(src, l...)
+    checkbounds(src, u...)
+    # If a disk array, cache the src so we don't read too many times
+    src_parent = isdisk(src) ? DiskArrays.cache(parent(src)) : parent(src)
+    @inbounds broadcast!(dst, CartesianIndices(dst)) do I
```

Comment:
would it be worthwhile to optionally run this threaded?

Reply:
Oh the reason not to is currently it works on GPU and DiskArrays as is, threading will break that

Reply:
well if threaded is false by default then you could either broadcast or do a threaded for loop. But after seeing just how fast this is I don't think that would only be relevant in some niche use cases